packages/components/src/internal/components/EditInlineField.tsx

```diff
@@ -1,36 +1,54 @@
 import React, { FC, FormEvent, memo, ReactNode, useCallback, useMemo, useReducer, useRef, useState } from 'react';
 import moment from 'moment';
+import classNames from 'classnames';
+import Formsy from 'formsy-react';
 
-import { getDateFormat } from '../util/Date';
-
+import { getColDateFormat, getDateFormat, getJsonDateTimeFormatString } from '../util/Date';
 import { Key, useEnterEscape } from '../../public/useEnterEscape';
-
 import { DateInput } from './DateInput';
 import { useServerContext } from './base/ServerContext';
-import classNames from 'classnames';
+import { QueryColumn } from '../../public/QueryColumn';
+import { resolveDetailEditRenderer } from './forms/detail/DetailEditRenderer';
 
 interface Props {
     allowBlank?: boolean;
     allowEdit?: boolean;
     className?: string;
+    column?: QueryColumn;
     emptyText?: string;
     label?: string;
     name: string;
     onChange?: (name: string, newValue: any) => void;
     placeholder?: string;
-    type: 'date' | 'text' | 'textarea';
-    value: any;
+    type: string;
+    value: any; // could be a primitive value or a RowValue (from internal/query/selectRows)
+    useJsonDateFormat?: boolean;
 }
 
 export const EditInlineField: FC<Props> = memo(props => {
-    const { allowBlank, allowEdit, className, emptyText, label, name, onChange, placeholder, type, value } = props;
+    const {
+        allowBlank,
+        allowEdit,
+        className,
+        emptyText,
+        label,
+        name,
+        onChange,
+        placeholder,
+        type,
+        value,
+        column,
+        useJsonDateFormat,
```

Comment:
I guess there's a good reason to need this, but *sigh*.

Reply:
Sigh indeed